lib/cylc/task_pool.py

```diff
@@ -244,8 +242,8 @@ def release_runahead_tasks(self):
 
         # Any finished tasks can be released immediately (this can happen at
         # restart when all tasks are initially loaded into the runahead pool).
-        for itask_id_maps in self.runahead_pool.values():
-            for itask in itask_id_maps.values():
+        for itask_id_maps in self.runahead_pool.copy().values():
```

Comment:
What is the purpose of the `copy()` here? `values`/`keys`/`items` all return lists.